app/src/main/java/com/example/plantsapp/presentation/ui/tasks/TasksViewModel.kt

```diff
@@ -1,23 +1,27 @@
 package com.example.plantsapp.presentation.ui.tasks
 
-import androidx.core.net.toUri
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.repository.PlantsRepository
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import java.util.Date
 
-class TasksViewModel : ViewModel() {
+class TasksViewModel(
+    private val repository: PlantsRepository
+) : ViewModel() {
 
-    val tasks: LiveData<List<Task>> = MutableLiveData(
-        listOf(
-            Task(WATERING_ACTION, "Hector", PLANT_PICTURE.toUri()),
-            Task(WATERING_ACTION, "Elvis", PLANT_PICTURE.toUri()),
-            Task(WATERING_ACTION, "Arthur", PLANT_PICTURE.toUri()),
-            Task(WATERING_ACTION, "Silvia", PLANT_PICTURE.toUri()),
-            Task(WATERING_ACTION, "Robert", PLANT_PICTURE.toUri()),
-        )
-    )
-}
+    private val _tasks: MutableLiveData<List<Task>> = MutableLiveData()
+    val tasks: LiveData<List<Task>> = _tasks
 
-private const val PLANT_PICTURE = "https://www.vippng.com/png/detail/41-414674_house-plant-png.png"
-private const val WATERING_ACTION = "Water"
+    init {
+        viewModelScope.launch {
+             repository.getTasksForDate(Date()).collect {
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html
https://stackoverflow.com/questions/28730136/should-i-use-java-util-date-or-switch-to-java-time-localdate
I think, lets try to use LocalDataTime here instead of Date